src/cmake/CheckLibcxxInUse.cmake

```diff
@@ -36,6 +36,21 @@ function(check_libcxx_in_use variable)
             message(CHECK_PASS "found")
         else()
             message(CHECK_FAIL "not found")
+            return()
+        endif()
+
+        message(CHECK_START "Checking if libc++ hacks are being used")
+        list(APPEND CMAKE_MESSAGE_INDENT "  ")
+
+        include(CheckCXXSourceCompiles)
+        check_cxx_source_compiles("#include<compare>\nusing _=std::compare_three_way;int main(){}" ${variable_no_hacks})
+        set(${variable_no_hacks} ${${variable_no_hacks}} PARENT_SCOPE)
+
+        list(POP_BACK CMAKE_MESSAGE_INDENT)
+        if(${variable_no_hacks})
+            message(CHECK_PASS "not used")
+        else()
+            message(CHECK_FAIL "used")
```

Comment:
Shouldn't we assume that libc++-14 is being used for clang-14? In such a case, a simple compiler version check would be enough. It would also simplify Conan's case a lot.

Reply:
Great idea. I used `CMAKE_CXX_COMPILER_VERSION`.